### ISSUE TITLE
Tooltip for twitter search's 'Filter Old' attribute.

### ISSUE THREAD

**Issue by rhuss** [syndesisio/syndesis-ui#970]
When selecting a "Twitter Search" action, there are tooltips on all fields except for 'Filter Old' field.

Besides this,  'Filter Old' seems a bit of an odd label (especially as a grey-haired coder, you hope that this kind of filter process still is some years away ;-P

![image](https://user-images.githubusercontent.com/99080/30537858-b36db718-9c6b-11e7-9a5f-7b7ada8faeca.png)

**Comment by rhuss** [comment]
Also, it seems that the order of the fields is not deterministic. In a second try I got:

![image](https://user-images.githubusercontent.com/99080/30543141-3fd93b62-9c82-11e7-9d3e-cad6b2837179.png)

**Comment by gashcrumb** [comment]
So the labels and tooltips shown are defined in the connector project.  We'll have to check if the checkbox template though exposes the tooltip like the input fields.

The order is based on what we get back from the rest API I believe but will double-check.

**Comment by rhuss** [comment]
I will check the descriptions in the connector definitions.

**Comment by rhuss** [comment]
@gashcrumb the definition of the filter property looks like:

```json
 "filterOld": {
                  "kind": "parameter",
                  "displayName": "Filter Old",
                  "group": "filter",
                  "label": "consumer,filter",
                  "required": false,
                  "type": "boolean",
                  "javaType": "boolean",
                  "tags":[],
                  "deprecated": false,
                  "secret": false,
                  "defaultValue": true,
                  "description": "Filter out old tweets that have previously been polled. This state is stored in memory only and is based on last tweet id."
                }
```

so, it looks like that the `description` is ignored for a tooltip. 

I will change the `displayName` though to something more clear.

**Comment by gashcrumb** [comment]
@rhuss thanks!  Would it make sense to also have the keyword search input as the first input?

Also so far the form is displayed in the same order from what I can see as the data coming back.  If you open the network tab and just click on different connections you can easily see the JSON response back.  My guess is normally it's alphabetical but it could be non-deterministic.  We could consider adding an additional parameter at the root of the property definition to allow connectors to specify a particular order of fields, right now there is none.

**Comment by gashcrumb** [comment]
Also looks like we pass the description on to the `hint` property when we're creating the checkbox but our current template for checkboxes doesn't appear to have that `?` icon like the regular inputs.

**Comment by rhuss** [comment]
@gashcrumb yes, I think the keyword field is the most important (and only mandatory) field, so having it first makes much sense.

I see, the properties are stored in a hash in the action definition (so no deterministic order). Wonder whether an array wouldn't be the better choice here ? Or an 'order' field as you suggest, but I find an array more intuitive. Alphabetically is not really an option as we loose the possibility to make a sensible grouping (like having keywords first).

Regardless how, the order should stay the same across requests.  

Also, wonder why 'keywords' is not marked mandatory. Fetching all tweets from twitter (by not specifying a keyword) doesn't make much sense, or ? So if no one cries, I'm going to mark `keywords` as required.

**Comment by gashcrumb** [comment]
Having a separate array that controls the order of form controls is nice because you can do things like:

`['control1', 'control2', '*']`

where any control not mentioned in the array can go into the `*` position, so as another example:

`['ImTotallyFirst', 'ImAlwaysSecond', '*', 'ImAlwaysLast']`

It worked nicely for hawtio-forms, which is kinda the base I've been using for our work here.

**Comment by gashcrumb** [comment]
We should open a separate issue to track this BTW

**Comment by rhuss** [comment]
But with the '*' option, wouldn't you still have the non-deterministic behaviour for the remaining keys ?

**Comment by gashcrumb** [comment]
Correct, however it also means you can impose some order but also be flexible and allow for changes in the form definition easily enough.

**Comment by rhuss** [comment]
looks good (but tbh I would nail the order down for everything ;-)

**Comment by gashcrumb** [comment]
Cool.  If it helps there's examples here of what I'm usually talking about when discussing forms -> http://forms.hawt.io/forms2/simple_form?main-tab=hawtio-forms2-tests-form2&sub-tab=hawtio-forms2-tests-form2-simple_form

**Comment by gashcrumb** [comment]
Closing this as the tooltip is there now...
